R/backport_linter.R

```diff
@@ -0,0 +1,82 @@
+#' @describeIn linters that checks for usage of unavailable functions. Not reliable for testing r-devel dependencies.
+#' @export
+backport_linter <- function(r_version = getRversion()) {
+  function(source_file) {
+    if (inherits(r_version, "numeric_version")) r_version <- format(r_version)
+    if (is.null(source_file$xml_parsed_content) || r_version >= "r") return(list())
```

Comment:
Taking another look, can't we just remove that condition?
It's never going to be true for any `r_version`.
Sorry for being slow.

Reply:
I think you're right because here:

```
1:(match(FALSE, r_version < names(backports), nomatch = length(backports)) - 1L)
```

 1. If the condition is mixed `TRUE`/`FALSE` that's normal
 2. If it's all `TRUE`, we use the `nomatch`
 3. If it's all `FALSE` we get back `1`

So the edge case I was guarding against is irrelevant as implemented.

I should replace it with a warning about trying to depend on an R version older than we've recorded.

Reply:
Sounds good. i.e. `numeric_version(r_version)[1L, 1L] < 3`